### ISSUE TITLE
Text styles can't be customized in high contrast mode

### ISSUE THREAD

**Issue by paperbackdragon** [microsoft/react-native-windows#10207]
### Problem Description

When a high contrast theme is active on Windows, if I have custom styles set on some text (such as a link), those styles don't show up in the UI. It will set all text to the same color and text-decoration, regardless.

Screenshot example - the "Get help", "Give feedback", and "Unlink this PC" text should all be a shade of blue, and should have underlines when hovered over, but they're all white, and nothing happens on hover:

<img width="1020" alt="Screen Shot 2022-06-27 at 1 27 04 PM" src="https://user-images.githubusercontent.com/899430/176029691-a9b8c78f-3837-4869-a099-2a236016371e.png">

### Steps To Reproduce

1. Make an app with a `<Text>` element and a `style` prop. Set the `color` to something bright and obvious.
2. Go into Windows 11 settings and choose the "Aquatic" high contrast theme
3. Go back to the app. The text will be white, no matter what color you chose.

### Expected Results

Ideally the text respects the styles applied to it.

### CLI version

0.66.0

### Environment

```markdown
System:
    OS: Windows 10 10.0.25131
    CPU: (20) x64 Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz
    Memory: 106.19 GB / 127.71 GB
  Binaries:
    Node: 16.9.1 - D:\CxCache\OneDrive.NodeJS.Windows.1.0.0\tools\nodejs\node.EXE
    Yarn: 1.22.11 - D:\CxCache\OneDrive.NodeJS.Windows.1.0.0\tools\nodejs\yarn.CMD
    npm: 7.21.1 - D:\CxCache\OneDrive.NodeJS.Windows.1.0.0\tools\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.17134.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.11.32602.291 (Visual Studio Enterprise 2019), 17.2.32602.215 (Visual Studio Enterprise 2022)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.66.0 => 0.66.0
    react-native-windows: 0.66.0 => 0.66.0
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2019

### Build Configuration

Debug

**Comment by asklar** [comment]
@chiaramooney could you take a look?

**Comment by chiaramooney** [comment]
Looking into this now!

**Comment by chiaramooney** [comment]
@paperbackdragon It looks like when backgroundColor is set the customizations to color (the foreground color of the text appear). Currently investigating into why this is the case, but for a temporary workaround setting `backgroundColor: 'transparent'` works.

**Comment by acoates-ms** [comment]
Possibly related to this:
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.highcontrastadjustment?view=winrt-22621#windows-ui-xaml-uielement-highcontrastadjustment

I think XAML assumes by default that apps do not properly handle high contrast.
